plugin/hover.py

```diff
@@ -366,3 +370,15 @@ def run_async() -> None:
                 session.run_code_action_async(actions[index], progress=True, view=self.view)
 
         sublime.set_timeout_async(run_async)
+
+    def try_open_custom_uri_async(self, href: str) -> None:
+        row, col_utf16 = row_col_from_uri_fragment(href)
+        if isinstance(row, int):
+            position = {"line": row, "character": col_utf16 or 0}  # type: Position
+            r = {"start": position, "end": position}  # type: Optional[Range]
+        else:
+            r = None
+        for session in self.sessions():
+            promise = session.open_uri_async(href, r)
+            if not promise.resolved or isinstance(promise.value, sublime.View):
+                return
```

Comment:
I'm a bit unsure about the correctness of this promise state check. I need to check whether a session will handle the URI. I check that by looking whether the promise is resolved immediately with a None result.

Reply:
It seems not that easy to handle that due to limitations of the current Promise implementation. If it supported `catch` then we could chain all and throw when current session has handled it.

But something like this might work (untested).

```py
        promise = Promise.resolve(False)
        for session in self.sessions():
            promise = promise.then(partial(self.try_open_next, session, href, r))

    def try_open_next(self, session: Session, href: str, r: Optional[Range], was_handled: bool) -> Promise[bool]:
        if was_handled:
            return Promise.resolve(True)
        return session.open_uri_async(href, r).then(lambda view: bool(view))
```

Reply:
I solved it by introducing another method `try_open_uri_async` that returns `None` when it can't handle the uri scheme.